codec/amino.go

```diff
@@ -34,6 +35,8 @@ func New() *Codec {
 // codec.
 func RegisterCrypto(cdc *Codec) {
 	cryptoamino.RegisterAmino(cdc)
+	cryptoamino.RegisterKeyType(protokeys.PrivateKey{}, "tendermint/proto/PrivateKey")
```

Comment:
@marbar3778 I'm getting some errors with the encoding because it doesn't recognize the new priv/pubkey. Can you take a look when you have time?

Reply:
you will have to define your own proto type. we decided to only support ed25519 in tendermints proto key types

Reply:
created a pr to show how this could be done simply: https://github.com/cosmos/cosmos-sdk/pull/5997

if this works i can work on completing that PR